drivers/soundwire/intel.h

```diff
@@ -47,15 +50,100 @@ struct sdw_intel {
 #endif
 };
 
-int intel_link_startup(struct auxiliary_device *auxdev);
-int intel_link_process_wakeen_event(struct auxiliary_device *auxdev);
+#define cdns_to_intel(_cdns) container_of(_cdns, struct sdw_intel, cdns)
 
-struct sdw_intel_link_dev {
-	struct auxiliary_device auxdev;
-	struct sdw_intel_link_res link_res;
-};
+#define INTEL_MASTER_RESET_ITERATIONS	10
+
+static inline void sdw_intel_debugfs_init(struct sdw_intel *sdw)
+{
+	if (sdw && sdw->link_res && sdw->link_res->hw_ops &&
+	    sdw->link_res->hw_ops->debugfs_init)
+		sdw->link_res->hw_ops->debugfs_init(sdw);
+}
+
+static inline void sdw_intel_debugfs_exit(struct sdw_intel *sdw)
+{
+	if (sdw && sdw->link_res && sdw->link_res->hw_ops &&
```

Comment:
the count of "sdw && sdw->link_res && sdw->link_res->hw_ops" is 11, can we define a macro ?

Reply:
done, thanks for the suggestion @RanderWang